### PR TITLE
Refactor/177 Flatten POKO hierarchy

### PR DESCRIPTION
**Pull request by bbjubjub2494**
For #177 I'm going to submit multiple PRs in increasing order of invasiveness.

This PR flattens the POKO hierarchy as recommended in the code review.

Note: It adds a ton of deprecation warnings. That's so as to minimize the amount of lines touched and potential conflicts. At some point where we have no PRs outstanding, we can auto-refactor those out.

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 import android.os.Parcelable
+import kotlinx.parcelize.Parcelize
 
 /**
- * An interface representing a category to which a CategorizedPicture can belong
- * id is used to uniquely identify a Category
+ * Class representing a category to which a CategorizedPicture can belong
+ *
+ * @param name the name of the category (case-sensitive)
```
Comment (wattsecs) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt]:
```suggestion
 * @property name the name of the category (case-sensitive)
```

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+ * @param id uniquely identifies the category
```
Comment (wattsecs) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt]:
```suggestion
 * @property id uniquely identifies the category
```

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt [hunk 3]
```diff
@@ -1,12 +1,14 @@
...
+ * @param representativePicture a categorized picture, can be null
```
Comment (wattsecs) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Category.kt]:
```suggestion
```

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Dataset.kt [hunk 4]
```diff
@@ -1,18 +1,20 @@
 package com.github.HumanLearning2021.HumanLearningApp.model
 
 import android.os.Parcelable
-
+import kotlinx.parcelize.Parcelize
 
 /**
- * Interface describing a Dataset.
+ * Representation of a Dataset.
  * @property id should be used to uniquely identify the Dataset
```
Comment (wattsecs) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/model/Dataset.kt]:
```suggestion
 * @property id uniquely identifies the Dataset
```

**Review by wattsecs — CHANGES_REQUESTED** [review]
Some typos/improvements for the kdoc

**Review by wattsecs — APPROVED** [review]

**Review by Jaakik — APPROVED** [review]

**Pull request merged**